### ISSUE TITLE
"Undefined index: properties" in LaravelToUML.php

### ISSUE THREAD

**Issue by orjanalmen** [andyabih/laravel-to-uml#1]
I get an error "Undefined index: properties" pointing to LaravelToUML and getSource()
which is:
```
foreach ($c['properties'] as $property) {
    $properties[] = $property->name;
}
foreach ($c['methods'] as $method) {
    $methods[] = $method->name . "()";
}
$source = "[{$c['name']}";
```
I tried to add ?? [] on the $c['properties'] so it would be empty if non-existing, but to no improvement, so I started investigated further and I found a big chunk of empty arrays in the $this->classes array, seemingly to be namespaces.

I added this as first clause in the foreach:
```            
if ($c === []) {
    continue;
}
```
And the code started working. That would be an workaround of course, just helping the symptoms

**Comment by andyabih** [comment]
Hiya! Do you have any classes with the string "App" instead of them? Maybe something like "AuditionApplicationController" for example?

Pushed a new release right now that will work for those.

**Comment by orjanalmen** [comment]
Nope. But now, I realize there are two sets of classes in the $this->classes array, one copy with "\App\..." and one with "App\...", and it's those with "App\..." that is empty

**Comment by andyabih** [comment]
Give it a try with the new 1.0.1 release. If it still doesn't work, would highly appreciate a dump of the `$this->classes` array if possible.

**Comment by orjanalmen** [comment]
No, It's not working.

I just took a sample here:
```
array:111 [▼
  "App\Actions/Fortify/CreateNewUser" => []
  "App\Actions/Fortify/PasswordValidationRules" => []
  "App\Actions/Fortify/ResetUserPassword" => []
  "App\Actions/Fortify/UpdateUserPassword" => []
  "App\Actions/Fortify/UpdateUserProfileInformation" => []
  "App\Actions/Jetstream/DeleteUser" => []
  "\App\Actions\Fortify\CreateNewUser" => array:5 [▶]
  "\App\Actions\Fortify\PasswordValidationRules" => array:5 [▶]
  "\App\Actions\Fortify\ResetUserPassword" => array:5 [▶]
  "\App\Actions\Fortify\UpdateUserPassword" => array:5 [▶]
  "\App\Actions\Fortify\UpdateUserProfileInformation" => array:5 [▶]
  "\App\Actions\Jetstream\DeleteUser" => array:5 [▶]
]
```

**Comment by andyabih** [comment]
Something tells me it might be related to the separators being either a forward slash or a backwards slash. I'll take a look at it and will update the issue here when resolved. Thanks for posting this.

**Comment by rowanfuchs** [comment]
I've got the same issue, I tried removing the models as some had the App you mentioned but it didn't do the trick. I've dumped classes, I hope it helps.

> array:60 [▼
>   "App\Channels/NotificationChannel" => []
>   "App\Events/ChatMessage" => []
>   "App\Http/Controllers/AuthController" => []
>   "App\Http/Controllers/ChatController" => []
>   "App\Http/Controllers/GroupController" => []
>   "App\Http/Controllers/IngredientController" => []
>   "App\Http/Controllers/LanguagesController" => []
>   "App\Http/Controllers/RecipeController" => []
>   "App\Http/Controllers/UserSettingsController" => []
>   "App\Jobs/TranslateIngredient" => []
>   "App\Jobs/TranslateRecipe" => []
>   "App\Jobs/TranslateStep" => []
>   "App\Mail/Contact" => []
>   "App\Mail/ResetPassword" => []
>   "App\Messages/NotificationMessage" => []
>   "App\Models/Chat" => []
>   "App\Models/Course" => []
>   "App\Models/Cuisine" => []
>   "App\Models/Difficulty" => []
>   "App\Models/Group" => []
>   "App\Models/GroupApplication" => []
>   "App\Models/GroupInvitation" => []
>   "App\Models/Ingredient" => []
>   "App\Models/IngredientMeasurement" => []
>   "App\Models/IngredientSorting" => []
>   "App\Models/Languages" => []
>   "App\Models/Recipe" => []
>   "App\Models/RecipeSteps" => []
>   "App\Models/User" => []
>   "App\Models/UserSettings" => []
>   "\App\Channels\NotificationChannel" => array:5 [▶]
>   "\App\Events\ChatMessage" => array:5 [▶]
>   "\App\Http\Controllers\AuthController" => array:5 [▶]
>   "\App\Http\Controllers\ChatController" => array:5 [▶]
>   "\App\Http\Controllers\GroupController" => array:5 [▶]
>   "\App\Http\Controllers\IngredientController" => array:5 [▶]
>   "\App\Http\Controllers\LanguagesController" => array:5 [▶]
>   "\App\Http\Controllers\RecipeController" => array:5 [▶]
>   "\App\Http\Controllers\UserSettingsController" => array:5 [▶]
>   "\App\Jobs\TranslateIngredient" => array:5 [▶]
>   "\App\Jobs\TranslateRecipe" => array:5 [▶]
>   "\App\Jobs\TranslateStep" => array:5 [▶]
>   "\App\Mail\Contact" => array:5 [▶]
>   "\App\Mail\ResetPassword" => array:5 [▶]
>   "\App\Messages\NotificationMessage" => array:5 [▶]
>   "\App\Models\Chat" => array:5 [▶]
>   "\App\Models\Course" => array:5 [▶]
>   "\App\Models\Cuisine" => array:5 [▶]
>   "\App\Models\Difficulty" => array:5 [▶]
>   "\App\Models\Group" => array:5 [▶]
>   "\App\Models\GroupApplication" => array:5 [▶]
>   "\App\Models\GroupInvitation" => array:5 [▶]
>   "\App\Models\Ingredient" => array:5 [▶]
>   "\App\Models\IngredientMeasurement" => array:5 [▶]
>   "\App\Models\IngredientSorting" => array:5 [▶]
>   "\App\Models\Languages" => array:5 [▶]
>   "\App\Models\Recipe" => array:5 [▶]
>   "\App\Models\RecipeSteps" => array:5 [▶]
>   "\App\Models\User" => array:5 [▶]
>   "\App\Models\UserSettings" => array:5 [▶]
> ]

**Comment by andyabih** [comment]
@orjanalmen @rowanfuchs Hi guys, was a bit hard trying to recreate it as for some reason it wasn't happening locally on my system. One of the projects I had did show an `undefined index: properties` finally, and hopefully it's due to the same reasons you are experiencing it.

I've added a fix, it was related to the preloaded files, so please let me know if it's now working in release 1.0.3. If not, I'll look into it further.

Thanks!

**Comment by orjanalmen** [comment]
This was fixed for me with 1.0.3

**Comment by rowanfuchs** [comment]
Same for me, what an amazing fast response and resolution. Thank you very much for this beautiful package.
